File: FE/lib/data/models/room.dart
import 'dart:io';

class Room {
  final int? id;
  final String? roomCode;
  final String? roomType;
  final String? roomName;
  final String? roomDescription;
  final double? roomPrice;
  final int? total_guest;
  final bool? booked;
  final String? photo;
  final bool? ac;
  final bool? tv;
  final bool? miniBar;
  final bool? jacuzzi;
  final bool? balcony;
  final bool? kitchen;
  final File? photoFile;

  Room({
    this.id,
    this.roomCode,
    this.roomName,
    this.roomDescription,
    this.roomPrice,
    this.total_guest,
    this.roomType,
    this.booked,
    this.photo,
    this.ac,
    this.tv,
    this.miniBar,
    this.balcony,
    this.jacuzzi,
    this.kitchen,
    this.photoFile,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'roomCode': roomCode,
      'roomName': roomName,
      'roomDescription': roomDescription,
      'booked': booked,
      'roomPrice': roomPrice,
      'roomType': roomType,
      'total_guest': total_guest,
      'ac': ac,
      'tv': tv,
      'miniBar': miniBar,
      'balcony': balcony,
      'jacuzzi': jacuzzi,
      'kitchen': kitchen,
      'photo': photo
    };
  }

  factory Room.fromJson(Map<String, dynamic> jsonData) {
    return Room(
      id: jsonData['id'] ?? 0,
      roomCode: jsonData['roomCode'] ?? "",
      roomName: jsonData['roomName'] ?? "",
      roomDescription: jsonData['roomDescription'] ?? "",
      roomPrice: (jsonData['roomPrice'] ?? 0).toDouble(),
      roomType: jsonData['roomType'] ?? "",
      total_guest: jsonData['total_guest'] ?? "",
      booked: jsonData['booked'] ?? false,
      ac: jsonData['ac'] ?? false,
      tv: jsonData['tv'] ?? false,
      miniBar: jsonData['miniBar'] ?? false,
      balcony: jsonData['balcony'] ?? false,
      jacuzzi: jsonData['jacuzzi'] ?? false,
      kitchen: jsonData['kitchen'] ?? false,
      photo: jsonData['photo'] ?? "",
    );
  }
}

File: BE/src/main/java/com/dailycodework/lakesidehotel/model/Room.java
package com.dailycodework.lakesidehotel.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import org.apache.commons.lang3.RandomStringUtils;

import java.math.BigDecimal;
import java.sql.Blob;
import java.util.ArrayList;
import java.util.List;

/**
 * @author Simpson Alfred
 */
@Entity
@Getter
@Setter
@AllArgsConstructor
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String roomCode;
    private String roomType;
    private String roomName;
    private BigDecimal roomPrice;
    private boolean isBooked = false;
    private boolean ac = false;
    private boolean tv = false;
    private boolean miniBar = false;
    private boolean balcony = false;
    private boolean jacuzzi = false;
    private boolean kitchen = false;
    private int total_guest;

    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String roomDescription;

    @Lob
    private Blob photo;

    @OneToMany(mappedBy = "room", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<BookedRoom> bookings;

    public Room() {
        this.bookings = new ArrayList<>();
    }

    public void addBooking(BookedRoom booking) {
        if (bookings == null) {
            bookings = new ArrayList<>();
        }
        bookings.add(booking);
        booking.setRoom(this);
        isBooked = true;
        String bookingCode = RandomStringUtils.randomNumeric(10);
        booking.setBookingConfirmationCode(bookingCode);
    }
}
